src/models/index.ts

```diff
@@ -30,3 +33,42 @@ export const DEFAULT_STATE: StateBase = {
   shelf: DEFAULT_SHELF_SPEC,
   result: DEFAULT_RESULT,
 };
+
+export interface SerializableState {
+  config: VoyagerConfig;
+  shelf: Shelf;
+  result: ResultIndex;
+  dataset: {
+    isLoading: boolean;
+    name: string;
+    data: Data
+  };
+  tableschema: TableSchema<FieldSchema>;
+}
+
+export function toSerializable(state: Readonly<StateBase>): SerializableState {
+  const asSerializable = {
+    config: state.config,
+    shelf: state.shelf,
+    result: state.result,
+    dataset: {
+      isLoading: state.dataset.isLoading,
+      name: state.dataset.name,
+      data: state.dataset.data,
+    },
+    tableschema: state.dataset.schema.tableSchema(),
+  };
+
+  return asSerializable;
+}
+
+export function fromSerializable(serializable: SerializableState): Readonly<StateBase> {
+  // We make a clone of this object to not modify the input param.
+  const deserialized = JSON.parse(JSON.stringify(serializable));
```

Comment:
We have `duplicate` in vega-lite's `util` which call parse & stringify, but better use consistent calls?